49.md

```diff
@@ -0,0 +1,88 @@
+not need. This either runs into silent failures or requires the user to understand what functions their wallet supports
+and what functions the app needs.
+
+Finally, Nostr Wallet Connect requires the wallet to generate both key pairs and the app to use these. This limits the
+key management strategies that an app can use and often leads to suboptimal key management.
+
+This NIP proposes a new protocol that solves these problems by having the wallet and app generate a NWC connection URI
+together. This URI is then used to connect the wallet and app.
+
+## Terms
+
+- **app**: Nostr app on any platform that wants to permissioned access to a user's lightning wallet.
+- **wallet**: A nostr wallet connect capable lightning wallet that the user wants to connect to the app.
+
+## Theory of Operation
+
+1. The user wants to connect their wallet to an app. The app provides a QR code or other means of connecting to the app.
+   The QR code contains a `nostr+walletauth://` URI that contains the app's public key, the relay to communicate over,
+   and a list of functions that the app requires.
+2. The user scans the QR code and their wallet prompts them the required functions and asks them to confirm that they
+   want to connect to the app.
+3. On confirmation, the wallet generates a new key pair and sends the public key to the app in a confirmation event.
+4. With the combination of both keys, the wallet and the app both will generate the same connection URI. They can now
+   use NWC to communicate.
+
+## Specification
+
+### URI Format
+
+The **app** generates this connection URI with protocol `nostr+walletauth:` and base path it's hex-encoded `pubkey`
+with the following query string parameters:
+
+- `relay` Required. URL of the relay where the **wallet** is connected and will be listening for events. May be
+  more than one.
+- `required_commands` Required. A space-separated list of commands that the **wallet** requires. The **wallet**
+  MUST NOT connect if it does not support all of these permissions.
+- `optional_commands` Optional. A space-separated list of commands that the **wallet** can enable to add additional
+  functionality. The **wallet** MAY ignore these.
+- `budget` Optional. A budget that the **wallet** will use to limit the amount of funds that the **app** can spend.
+  The **app** MUST NOT spend more than this amount. The budget is a string of the form `max_amount/period`.
+  The `period` is one of `day`, `week`, `month`, `year`. The `max_amount` is a satoshi amount.
+
+#### Example URI
+
+```sh
+nostr+walletauth://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&required_commands=pay_invoice%20pay_keysend%20make_invoice%20lookup_invoice&optional_commands=list_invoices%20list_payments&budget=10000%2Fday
+```
+
+### Event
+
+This NIP only uses one event kind `13193`. We use a replaceable event to allow for the wallet and app to renegotiate
```

Comment:
> There is no requirement saying that one npub must host only one wallet.

yeah but that shouldn't be done, can add a note about that

> This should be a replaceable event.

it is one

> There is also the issue that an attacker can monitor for all events of this kind, and then create their own event to trick the client into thinking that the attacker pubkey is the wallet in a denial of service attack. This can be fixed by including a secret in the connection URL, and sending that in the encrypted payload

Good idea